### PR TITLE
HL API Allow document download

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Similar to the legacy API, sending a request to the `/Management/Document/{id}` endpoint with an `Accept` header of `application/octet-stream` will tell the server to respond with the file contents instead of the metadata.
Sending a HEAD request with the `Accept` header for downloading a document will respond with the correct headers but without the content.

`Toolbox::sendFile` was modified to accept a new parameter to allow the returning of a Response object rather than having the headers and content sent directly.

### CODE REVIEW

File: src/Toolbox.php [hunk 1]
```diff
@@ -584,35 +593,51 @@ public static function sendFile($file, $filename, $mime = null, $expires_headers
...
-            rawurlencode($filename)
-        );
-        header("Content-type: " . $mime);
+            rawurlencode($filename);
+        $headers['Content-Disposition'] = $content_disposition;
+        $headers['Content-type'] = $mime;
 
        // HTTP_IF_NONE_MATCH takes precedence over HTTP_IF_MODIFIED_SINCE
        // http://tools.ietf.org/html/rfc7232#section-3.3
+        $matches_cache = true;
```
Comment (cedric-anne) [inline src/Toolbox.php:615]:
```suggestion
        $matches_cache = false;
```

File: src/Toolbox.php [hunk 2]
```diff
@@ -584,35 +593,51 @@ public static function sendFile($file, $filename, $mime = null, $expires_headers
...
         if (isset($_SERVER['HTTP_IF_NONE_MATCH']) && trim($_SERVER['HTTP_IF_NONE_MATCH']) === $etag) {
-            http_response_code(304); //304 - Not Modified
+            $matches_cache = false;
```
Comment (cedric-anne) [inline src/Toolbox.php:617]:
```suggestion
            $matches_cache = true;
```

File: src/Toolbox.php [hunk 3]
```diff
@@ -584,35 +593,51 @@ public static function sendFile($file, $filename, $mime = null, $expires_headers
...
+        } else if (isset($_SERVER['HTTP_IF_MODIFIED_SINCE']) && @strtotime($_SERVER['HTTP_IF_MODIFIED_SINCE']) >= $lastModified) {
+            $matches_cache = false;
```
Comment (cedric-anne) [inline src/Toolbox.php:619]:
```suggestion
            $matches_cache = true;
```

File: src/Toolbox.php [hunk 4]
```diff
@@ -584,35 +593,51 @@ public static function sendFile($file, $filename, $mime = null, $expires_headers
...
+        }
+        if (!$matches_cache) {
```
Comment (cedric-anne) [inline src/Toolbox.php:621]:
```suggestion
        if ($matches_cache) {
```

**Review by cedric-anne — COMMENTED** [review]
`304` is sent when file `etag`/`last modified date` is matching the last known value found in browser cache.

**Review by orthagh — APPROVED** [review]

**Pull request merged**